src/main/java/net/mcreator/blockly/java/blocks/TextTrimBlock.java

```diff
@@ -22,21 +22,33 @@
 import net.mcreator.blockly.BlocklyCompileNote;
 import net.mcreator.blockly.BlocklyToCode;
 import net.mcreator.blockly.IBlockGenerator;
+import net.mcreator.blockly.java.ProcedureCodeOptimizer;
 import net.mcreator.generator.template.TemplateGeneratorException;
 import net.mcreator.ui.init.L10N;
 import net.mcreator.util.XMLUtil;
 import org.w3c.dom.Element;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class TextTrimBlock implements IBlockGenerator {
 	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
 
 		Element element = XMLUtil.getFirstChildrenWithName(block, "value");
 		if (element != null) {
-			master.append("(");
-			master.processOutputBlockWithoutParentheses(element);
-			master.append(").trim()");
+			String elementcode = BlocklyToCode.directProcessOutputBlock(master, element);
+			if (master.getTemplateGenerator() != null) {
+				if (master.getTemplateGenerator().hasTemplate("_trim.java.ftl")) {
+					Map<String, Object> dataModel = new HashMap<>();
+					dataModel.put("value", elementcode);

```

Comment:
Maybe rename this key to `input$value` as this is passed to procedure block template?

Reply:
I'd say it's fine as is now